### PR TITLE
fix: change gas limit fallback

### PR DESCRIPTION
**Pull request by OGPoyraz**
## Explanation

This PR reduce gas limit fallback from 95% to 35% of the block gas limit to improve how we handle failed gas limit estimations.

## References

Fixes: https://github.com/MetaMask/metamask-extension/issues/19692

### `@metamask/transaction-controller`

- **CHANGED**: Reduce gas limit fallback from 95% to 35% of the block gas limit on failed gas limit estimations

## Checklist

- [X] I've updated the test suite for new or updated code as appropriate
- [X] I've updated documentation (JSDoc, Markdown, etc.) for new or updated code as appropriate
- [X] I've highlighted breaking changes using the "BREAKING" category above as appropriate
- [X] I've prepared draft pull requests for clients and consumer packages to resolve any breaking changes

### CODE REVIEW

File: packages/transaction-controller/src/utils/gas.ts [hunk 1]
```diff
@@ -96,7 +96,7 @@ export function addGasBuffer(
   multiplier: number,
 ) {
   const estimatedGasBN = hexToBN(estimatedGas);
-  const maxGasBN = hexToBN(blockGasLimit).muln(0.9);
+  const maxGasBN = hexToBN(blockGasLimit).muln(0.3);
```
[the remark this answers is not in the document]
Comment (OGPoyraz) [reply packages/transaction-controller/src/utils/gas.ts]: This was not an intended change, removed it.

File: packages/transaction-controller/src/utils/gas.ts [hunk 2]
```diff
@@ -60,7 +60,7 @@ export async function estimateGas(
   const gasLimitBN = hexToBN(gasLimitHex);
 
   request.data = data ? add0x(data) : data;
-  request.gas = BNToHex(fractionBN(gasLimitBN, 19, 20));
```
[the remark this answers is not in the document]
Comment (OGPoyraz) [reply packages/transaction-controller/src/utils/gas.ts:63]: Done

**Review by vinistevam — DISMISSED** [review]

File: packages/transaction-controller/src/utils/gas.ts [hunk 3]
```diff
@@ -136,8 +132,12 @@ async function getGas(
     request.ethQuery,
   );
 
-  if (isCustomNetwork) {
-    log('Using original estimate as custom network');
+  if (isCustomNetwork || simulationFails) {
```
Comment (matthewwalsh0) [inline packages/transaction-controller/src/utils/gas.ts:135]: Does this mean we were buffering the fallback also? So effectively using 150% of 95% of the block gas limit...

File: packages/transaction-controller/src/utils/gas.ts [hunk 4]
```diff
@@ -136,8 +132,12 @@ async function getGas(
...
+    log(
+      isCustomNetwork
+        ? 'Using original estimate as custom network'
+        : 'Using original estimate as simulation failed',
```
Comment (matthewwalsh0) [inline packages/transaction-controller/src/utils/gas.ts:139]: Should this read `fallback estimate` for clarity?

**Review by matthewwalsh0 — DISMISSED** [review]

File: packages/transaction-controller/src/utils/gas.ts (hunk 3 above)
Comment (OGPoyraz) [reply packages/transaction-controller/src/utils/gas.ts:135]: Exactly that, it was not taking into account if the gas estimation failed or not.

File: packages/transaction-controller/src/utils/gas.ts (hunk 4 above)
Comment (OGPoyraz) [reply packages/transaction-controller/src/utils/gas.ts]: Done

**Review by matthewwalsh0 — APPROVED** [review]

**Pull request merged**